# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1517585840 18000
#      Fri Feb 02 10:37:20 2018 -0500
# Node ID 5c1d6ea818cdc506f21a45b6f72137c1ac71cc7f
# Parent  13c207835e797d1565005d9ec1780a138d44cfb0
fix regression in formatted output introduced in d4dd741b2794

* pr-output.cc (octave_print_internal (std::ostream&, double, bool)):
Restore call to set_format.
(octave_print_internal (std::ostream&, const Complex&, bool)):
Likewise.

diff --git a/libinterp/corefcn/pr-output.cc b/libinterp/corefcn/pr-output.cc
--- a/libinterp/corefcn/pr-output.cc
+++ b/libinterp/corefcn/pr-output.cc
@@ -1624,16 +1624,17 @@ octave_print_internal (std::ostream&, ch
 {
   panic_impossible ();
 }
 
 void
 octave_print_internal (std::ostream& os, double d,
                        bool pr_as_read_syntax)
 {
+  set_format (d);
   octave_print_internal (os, curr_float_display_fmt, d, pr_as_read_syntax);
 }
 
 void
 octave_print_internal (std::ostream& os, const float_display_format& fmt,
                        double d, bool pr_as_read_syntax)
 {
   if (pr_as_read_syntax)
@@ -2046,16 +2047,17 @@ pr_plus_format<> (std::ostream& os, cons
   else
     os << 'c';
 }
 
 void
 octave_print_internal (std::ostream& os, const Complex& c,
                        bool pr_as_read_syntax)
 {
+  set_format (c);
   octave_print_internal (os, curr_float_display_fmt, c, pr_as_read_syntax);
 }
 
 extern void
 octave_print_internal (std::ostream& os, const float_display_format& fmt,
                        const Complex& c, bool pr_as_read_syntax)
 {
   if (pr_as_read_syntax)
